experimental/lib/Transforms/Speculation/HandshakeSpeculation.cpp

```diff
@@ -337,21 +350,49 @@ std::optional<Value> findControlInputToBB(Operation *op) {
   handshake::FuncOp funcOp = op->getParentOfType<handshake::FuncOp>();
```

Comment:
just reading this after jiahui's comment, getting the func op should be removed right?

Reply:
yeah I agree the funcOp is not relevant here

Reply:
The problem is that `funcOp` is extensively used within this method.
Furthermore `targetBB` cannot be identified solely from `funcOp`.

I guess the name `findControlInputToBB` is misleading.
How about `findTriggerInputToSpeculator` and keeping the arguments?

Reply:
Yeah you are right, in any case, we need to iterate through the ops inside the funcOp. I think it looks nicer to determine the funcOp outside this method and pass it as an argument (so the method has two arguments: `handshake::FuncOp funcOp, unsigned BB`).